src/documentation/literal.rs

```diff
@@ -162,6 +162,18 @@ impl TryFrom<(&str, proc_macro2::Literal)> for TrimmedLiteral {
         // Either cut off `]` or `\n` - we don't need either.
         span.end.column = span.end.column.saturating_sub(1);
 
+        // If the line ending has more than one character, we have to account
+        // for that. Otherwise cut of the last character of the ending such that
+        // we can't properly detect them anymore.
+        if crate::reflow::extract_delimiter(content)
```

Comment:
we should consider detecting the delimiter from `rendered` (extracted blow) and fallback on the entire `content` as stage 2 and stage 3 use the default `\n`. This is mostly a perf thing avoiding to traverse the full content multiple times and as a fix for files that has multiple different line endings.

Reply:
That was my first attempts but the last character is dropped in the extraction of `rendered`. Unfortunately, we have to traverse content here.

Reply:
I guess technically we could convert the `Span` to a `Range` which we can extend easily by 2 characters and compare the last two to all known line endings or comment closing characters to determine the type. But that won't give us any performance edge since the conversion currently traverses the whole `content` again, without any caching done. Maybe adding a `Option<HashMap<..>` would be viable and introducing `Content` type which implements `Deref<str>`. But that's a separate topic.
I also did not take into account comments which do not contain a linewrap, so that would always fail.

Mea culpa, good as is :+1: